src/graphql/candyMachine.ts

```diff
@@ -15,24 +15,223 @@ import fs from 'fs/promises';
+  ammanMockStorage,
+  AmmanMockStorageDriver,
+} from '@metaplex-foundation/amman-client';
+import exec from 'await-exec';
 
 const dirname = path.resolve();
-const storageDir = "/app/tmp";
+
+const storageDir = process.env.APP_ENV === 'development' ? '/app/tmp' : 'tmp';
+
+// Downloads a zip file from zipUrl and returns the directory where zipUrl was unpacked
+const downloadZip = async (
+  zipUrl: string,
+  processId: string,
+): Promise<string> => {
+  // Unpack zip file
+  const processDir = `${storageDir}/${processId}`;
+  const zipFilesDir = `${storageDir}/${processId}/files`;
+  const zipFile = `${storageDir}/${processId}/files.zip`;
+  const dirExists = await fs
+    .stat(processDir)
+    .then(() => true)
+    .catch(() => false);
+
+  if (!dirExists) {
+    await mkdirp(`${storageDir}/${processId}`);
+    await download(zipUrl, zipFile);
+    await unzip(zipFile, zipFilesDir);
+  } else {
+    throw Error('zip dir already exists');
+  }
+
+  return zipFilesDir;
+};
+
+const SUPPORTED_MEDIA_FILE_EXTENSIONS = ['.png', '.jpg', '.jpeg', '.mp4'];
+
+const contentTypeForFileName = (
+  fileName: string,
+  supportedFileTypes = SUPPORTED_MEDIA_FILE_EXTENSIONS,
+) => {
+  const extensionToType = {
+    '.png': 'image/png',
+    '.jpg': 'image/jpeg',
+    '.jpeg': 'image/jpeg',
+    '.mp4': "video/mp4'",
+  };
+
+  const extension = fileName.split('.')[-1];
+  return extensionToType[extension];
+};
+
+const runUploadV2UsingHiddenSettings = async (
+  logger: winston.Logger,
+  processId: string,
+  args: {
+    collectionMint: string;
+    config: any;
+    callbackUrl: null | string;
+    guid?: string;
+    keyPair: string;
+    env: string;
+    filesZipUrl: string;
+    rpc: string;
+    setCollectionMint: boolean;
+  },
+) => {
+  const {
+    collectionMint: collectionMintParam,
+    setCollectionMint,
+    filesZipUrl,
+    config,
+    rpc,
+    env,
+    keyPair,
+  } = args;
+
+  logger.info('Uploading Candy Machine with hidden settings');
+
+  // setup connection and metaplex client
+  const bytes = bs58.decode(keyPair);
+  const walletKeyPair = web3.Keypair.fromSecretKey(Uint8Array.from(bytes));
+  const connection = new Connection(rpc);
+  const metaplex = new Metaplex(connection);
+  metaplex.use(keypairIdentity(walletKeyPair));
+  if (env === 'localnet') {
+    metaplex.use(ammanMockStorage('amman-mock-storage'));
+  } else {
+    if (config['storage'] !== 'nft-storage' || !config['nftStorageKey']) {
+      const message =
+        'hidden settings currently only works with nft storage and requires an nftStorageKey';
+      logger.error(message);
+      throw Error(message);
+    }
+
+    metaplex.use(
+      nftStorage({
+        token: config['nftStorageKey'],
+      }),
+    );
+  }
+
+  const storageDriver = metaplex.storage();
+
+  // First we need to upload the necessary NFT files from the zip
+  const zipFilesDir = await downloadZip(filesZipUrl, processId);
+  const templateNftPath = path.join(zipFilesDir, '0.json');
```

Comment:
I've seen a lot of duplicate usages of key strings, like `0.json`, `file_upload.zip`, etc. These should ideally be defined in a single place and imported or in some cases wrapped in simple functions.

Reply:
done